### PR TITLE
Cult tables now work like tables, can store cult paraphernalia SPOOKILY.

### PR DESCRIPTION
**Pull request by harmonyn**
- Fixes https://github.com/tgstation/-tg-station/issues/12249.
- Cult tables (the one the librarian has) and altars (the one that can be spawned or sometimes shows up on the asteroid) are now repathed to be children of tables, giving them all table functionality. I have a nagging feeling this is going to fuck something up, let me know if it does.
- Cult tables can effectively hide one tome and/or talisman in a spooky eldritch compartment, with fluff text.
- Edits all maps containing the above cult structures with the new path.

### CODE REVIEW

File: code/game/gamemodes/cult/cult_structures.dm [hunk 1]
```diff
@@ -3,32 +3,57 @@
...
+
+/obj/structure/table/cult/attack_hand(mob/user)
+	if(!occupied)
+		user << "<span class='notice'>You can't seem to find any drawers. Strange...</span>"
+		return
+	else
+		occupied = 0
+		hidden.loc = user.loc
+		user.put_in_hands(hidden)
+		user << "<span class='notice'>As you touch the desk, [hidden] flies out of a compartment that wasnn't there before and into your hands.</span>"
```
Comment (PKPenguin321) [inline code/game/gamemodes/cult/cult_structures.dm]: >wasnn't

Comment (duncathan) [inline code/game/gamemodes/cult/cult_structures.dm]: Also in the message for when you put something in, there's a comma splice; you should replace the comma with a semicolon

**Comment by bgobandit** [comment]
grammatical errors fixed, fuck my keyboard

also I changed the empty-hand message to more directly hint to the player that it might be storage

**Comment by Bawhoppen** [comment]
thank you bgo, i have wanted this for years

**Comment by Anonus** [comment]
You need a frame of some sort, so it doesn't produce the default frame(because that'd look silly), the framestack and buildstack vars to both be runed metal(it'd produce metal right now), and the buildstackamount and framestackamount vars to match up with the amounts used to build these(or slightly less)

Also, it'd be good to have these under one path so you don't need to do anything other than change the buildstackamount and framestackamount vars for them.

**Comment by duncathan** [comment]
Alternatively you could just make it do nothing on attack_by with a wrench

**Comment by Anonus** [comment]
Explosions and swarmers, though.

File: code/game/gamemodes/cult/cult_structures.dm [hunk 2]
```diff
@@ -3,32 +3,57 @@
...
 
-/obj/structure/cult/tome
-	name = "desk"
-	desc = "A desk covered in arcane manuscripts and tomes in unknown languages. Looking at the text makes your skin crawl"
+/obj/structure/table/cult
+	name = "mysterious desk"
+	desc = "A desk covered in arcane manuscripts and tomes in unknown languages. Looking at the text makes your skin crawl."
 	icon_state = "tomealtar"
 	luminosity = 1
+	var/occupied = 1
```
Comment (RemieRichards) [inline code/game/gamemodes/cult/cult_structures.dm]: this var is useless, 1000% unneeded.
just check for ```if(hidden)``` where you would ```if(occupied)```

**Comment by Bawhoppen** [comment]
Wooden table frame, then you put sandstone onto it

**Comment by bgobandit** [comment]
I'd prefer to make them not buildable, as A) it wouldn't fit the lore to mass-produce cult tables and B) it WILL be powergamed to allow cultists to hide infinite tomes and shit

**Comment by Bawhoppen** [comment]
I mean, if it had to be buildable something like that

**Comment by Razharas** [comment]
@bgobandit

**Comment by bgobandit** [comment]
holding off on resolving the merge conflict until the latest batch of map edits are merged; otherwise I'll just have to do it all over again

**Comment by Anonus** [comment]
Making infinite tables to store tomes isn't even practical because it doesn't have a cult check for removing them, so people can just.... click the table.

Also this is very not ready codewise, as all of the frame and build amount stuff is still an issue. You asked if making them tables would fuck stuff up you're getting an answer.
Also making it drop runed metal isn't an issue for building cult stuff because noncultists can't USE it.

**Comment by duncathan** [comment]
@bgobandit

**Comment by bgobandit** [comment]
obsolete due to new cult changes

**Pull request closed without merging**